libs/template/file.go

```diff
@@ -97,5 +118,35 @@ func (f *inMemoryFile) PersistToDisk() error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(path, f.content, f.perm)
+
+	if runsOnDatabricks() && f.isNotebook() {
+		return writeNotebook(path, f.content)
+	} else {
+		return os.WriteFile(path, f.content, f.perm)
+	}
+}
+
+func runsOnDatabricks() bool {
+	return os.Getenv("DATABRICKS_RUNTIME_VERSION") != ""
+}
+
+func writeNotebook(path string, content []byte) error {
+	if !strings.HasPrefix(path, "/Workspace/") {
+		return os.WriteFile(path, content, 0644)
+	} else {
+		w, err := databricks.NewWorkspaceClient()
```

Comment:
Default constructor will work because this code path is only executed on a workspace

Reply:
We expect the workspace client to be already configured and set in the context. You can use `root.WorkspaceClient(ctx)`  to get the workspace client.

Reply:
If we're going to depend on workspace context here, could you add an in-code comment?

Reply:
Changed it to `root.WorkspaceClient(ctx)`